content/exchange/artifacts/Generic.Collection.UAC.yaml

```diff
@@ -25,35 +25,38 @@ parameters:
 
 sources:
     - query: |
-        // fetch .tar.gz package
+        // fetch uac .tar.gz package
         LET uac_package <= SELECT * FROM Artifact.Generic.Utils.FetchBinary(ToolName="uac", IsExecutable=FALSE)
         // create temp dir
         LET temp_dir <= tempdir(remove_last=true)
         // uncompress the .tar.gz container
-        LET uncompress_tar_gz <= SELECT * FROM execve(argv=['tar', 'zxf', uac_package.FullPath[0]], cwd=temp_dir)
+        LET uncompress_tar_gz <= SELECT * FROM execve(argv=['tar', 'zxf', uac_package.OSPath[0]], cwd=temp_dir)
         // search for the correct uac source directory name
-        LET uac_source_directory <= SELECT FullPath FROM glob(globs=["uac-*"], root=temp_dir) WHERE IsDir = true
+        LET uac_source_directory <= SELECT OSPath FROM glob(globs=["uac-*"], root=temp_dir) WHERE IsDir = true
         // run uac
         LET run_uac <= SELECT * FROM execve(argv=[
                                                 "/bin/sh",
                                                 "-c",
                                                 "./uac -u " + CommandLineOptions + " ."
                                             ],
-                                            cwd=uac_source_directory.FullPath[0],
+                                            cwd=uac_source_directory.OSPath[0],
                                             sep="\n",
                                             length=2048
                                         )
-        // find the output files
-        LET find_output_files <= SELECT FullPath FROM glob(globs=["uac-*.log", "uac-*.tar.gz"], root=uac_source_directory.FullPath[0])
-        // upload the output files
-        LET upload_output_files <= SELECT upload(accessor="file", file=FullPath) AS Upload FROM find_output_files
+        // remove uac package from client
+        LET remove_uac_package <= SELECT rm(filename=uac_package.OSPath[0]) FROM scope()
```

Comment:
You generally don't need to explicitly delete the files if you use tempdir to get a temporary directory and get the tool to write there

Reply:
You are right. Actually, I have added TemporaryOnly=TRUE to the FetchBinary and removed the rm call.